### PR TITLE
Workflow automation post

### PR DESCRIPTION
**Pull request by singles**
Post about automation at hc - history, liam, probot.

### CODE REVIEW

File: _posts/2018-04-04-workflow-automation.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+* you will never forget to do something, as it's not your job anymore to remember
+* you will not make an error (besides dozen of errors which you made while writing such a script)
+* process is documented - very often code serves as a documentation
+* depends on a task, it sometimes comes with the coolness factor (see donut example above)
+
+In today's world, where more and more products provide an API, options are often limited only by our imagination. We have [IFTTT](https://ifttt.com/) and [Zapier](https://zapier.com/). We have self-hosted tools like [Kibitzr](https://kibitzr.github.io/) or [Beehive](https://github.com/muesli/beehive). This post focuses on "what" we automate at HolidayCheck and how we do it. 
+
+## First? Clean
+
+Each time a branch is created/updated in our repository, the content of this branch is deployed to staging environment (based on Mesos + Marathon). It means that a couple of minutes after pushing changes developer can show changes to the product owner. This allows us to iterate quickly.
```
Comment (Arturszott) [inline _posts/2018-04-04-workflow-automation.md]: "after pushing changes developer can show **them**"

Comment (Arturszott) [reply _posts/2018-04-04-workflow-automation.md]: "This allows us to iterate quickly." ->This enables fast software delivery.

File: _posts/2018-04-04-workflow-automation.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+On the other side, amount of resources (RAM + CPU) is limited. And people can sometimes forget to undeploy a branch when a feature is merged to production. That's how `deployment-cleaner` was born, created by Mathias.
```
Comment (Arturszott) [inline _posts/2018-04-04-workflow-automation.md]: `Mathias` link to him?

File: _posts/2018-04-04-workflow-automation.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+Simple. Effective. One problem that we had with mentioned event approach was, that in case of some bug/network issue webhook or request to the API might not be delivered, which would mean that without manual intervention, it would stay on staging environment "forever". 
+
+So, instead of 
+
+> after a branch is merged, undeploy this specific branch
+
+we wanted something more like
+
+> each X minutes, fetch a list of deployed branches from Marathon, compare then with branches on GitHub and undeploy those, which are no longer on GitHub
```
Comment (Arturszott) [inline _posts/2018-04-04-workflow-automation.md]: `each` or `every`?

File: _posts/2018-04-04-workflow-automation.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+![issue-prioritizer](img/posts/2018-04-04-workflow-automation/issue-prioritizer.png)
+
+
+Visible differences in `probot` : 
+
+* it utilizes [Github Apps API](https://developer.github.com/apps/) 
+* it's more opinionated, therefore easier - uses [@octokit/rest.js](https://github.com/octokit/rest.js) as GitHub API library, while in case of `liam` each task can use different client or dependencies
+* API calls are already repository aware (eg. you don't need to specify repository name and owner when you want to make a comment under PR, as this is prepopulated)
+* doesn't support scheduled tasks
```
Comment (lo1tuma) [inline _posts/2018-04-04-workflow-automation.md]: It does support this, but it is a [separate package](https://github.com/probot/scheduler).

**Pull request merged**